### PR TITLE
Add preliminary v0.2 event spec support

### PR DESCRIPTION
**Pull request by berndtj**
* maintain compatibility with 0.1
* copied over v01 to v02 and updated struct and tests to support

### CODE REVIEW

File: v02/event.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+// Event implements the the CloudEvents specification version master (0.2)
+// https://github.com/cloudevents/spec/blob/master/spec.md
+type Event struct {
+	// SpecVersion is a mandatory property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#specversion
+	SpecVersion string `json:"specversion" cloudevents:"CE-SpecVersion,required"`
+	// EventType is a mandatory property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#eventtype
+	Type string `json:"type" cloudevents:"CE-Type,required"`
```
Comment (matzew) [inline v02/event.go]: see the spec, this is `ce-type` etc

File: v02/event.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	Source string `json:"source" cloudevents:"CE-Source,required"`
+	// EventID is a mandatory property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#eventid
+	ID string `json:"id" cloudevents:"CE-ID,required"`
+	// EventTime is an optional property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#eventtime
+	Time *time.Time `json:"time,omitempty" cloudevents:"CE-Time"`
+	// SchemaURL is an optional property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#schemaurl
+	SchemaURL string `json:"schemaurl,omitempty" cloudevents:"CE-SchemaURL"`
```
Comment (matzew) [inline v02/event.go]: `ce-schemaurl`

File: v02/event.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+	SpecVersion string `json:"specversion" cloudevents:"CE-SpecVersion,required"`
+	// EventType is a mandatory property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#eventtype
+	Type string `json:"type" cloudevents:"CE-Type,required"`
+	// EventTypeVersion is an optional property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#eventtypeversion
+	Source string `json:"source" cloudevents:"CE-Source,required"`
+	// EventID is a mandatory property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#eventid
+	ID string `json:"id" cloudevents:"CE-ID,required"`
```
Comment (matzew) [inline v02/event.go]: basically all lowercase (e.g. `ce-id`)

File: v02/event.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	// https://github.com/cloudevents/spec/blob/master/spec.md#schemaurl
+	SchemaURL string `json:"schemaurl,omitempty" cloudevents:"CE-SchemaURL"`
+	// ContentType is an optional property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#contenttype
+	ContentType string `json:"contenttype,omitempty" cloudevents:"Content-Type"`
+	// Data is an optional property
+	// https://github.com/cloudevents/spec/blob/master/spec.md#data-1
+	Data interface{} `json:"data,omitempty" cloudevents:",body"`
+	// extension an internal map for extension properties not defined in the spec
+	extension map[string]interface{}
```
Comment (matzew) [inline v02/event.go]: extensions are optional, but it's different now, in 0.2 spec version

**Comment by tenczar** [comment]
These changes have been incorporated into a more complete change for v0.2. See #8

**Pull request closed without merging**